pkg/cloudprovider/aws/apis/v1alpha1/provider_validation.go

```diff
@@ -44,16 +43,9 @@ var (
 	securityGroupRegex = regexp.MustCompile("sg-[0-9a-z]+")
 )
 
-func (a *AWS) Validate(provisioner v1alpha5.Provisioner) (errs *apis.FieldError) {
+func (a *AWS) Validate() (errs *apis.FieldError) {
 	return errs.Also(
 		a.validate().ViaField("provider"),
-		a.validateProvisioner(provisioner),
-	)
-}
-
-func (a *AWS) validateProvisioner(provisioner v1alpha5.Provisioner) (errs *apis.FieldError) {
```

Comment:
This semantic doesn't work with ProviderRef. We can't have a validating webhook validate the AWSNodeTemplate CRD based on `kubeletConfiguration` which is a property of the Provisioner CRD. 

This check was pretty shallow in any case (making sure you don't pass `dockerd` with Bottlerocket so I don't think we lose too much by removing this).